roms_tools/plot.py

```diff
@@ -597,13 +631,464 @@ def _add_field_to_ax(
+            depth = depth.isel(s_rho=s)
+        return depth.load()
+
+    if compute_layer_depth:
+        layer_depth = _compute_and_slice_depth("layer")
+
+    if compute_interface_depth:
+        interface_depth = _compute_and_slice_depth("interface")
+
+    # Regrid laterally
+    if lat is not None or lon is not None:
+        if lat is not None:
+            lats = [lat]
+            title = title + f", lat = {lat}°N"
+        else:
+            resolution = infer_nominal_horizontal_resolution(grid_ds)
+            lats = _generate_coordinate_range(
+                field.lat.min().values, field.lat.max().values, resolution
+            )
+        lats = xr.DataArray(lats, dims=["lat"], attrs={"units": "°N"})
+
+        if lon is not None:
+            lons = [lon]
+            title = title + f", lon = {lon}°E"
+        else:
+            resolution = infer_nominal_horizontal_resolution(grid_ds, lat)
+            lons = _generate_coordinate_range(
+                field.lon.min().values, field.lon.max().values, resolution
+            )
+        lons = xr.DataArray(lons, dims=["lon"], attrs={"units": "°E"})
+
+        target_coords = {"lat": lats, "lon": lons}
+        lateral_regrid = LateralRegridFromROMS(field, target_coords)
+        field = lateral_regrid.apply(field).squeeze()
+
+        if compute_layer_depth:
+            layer_depth = lateral_regrid.apply(layer_depth).squeeze()
+
+    # Assign depth as coordinate
+    if compute_layer_depth:
+        field = field.assign_coords({"layer_depth": layer_depth})
+
+    if lat is not None:
+        field, layer_depth = _remove_edge_nans(
+            field, "lon", layer_depth if "layer_depth" in locals() else None
+        )
+    if lon is not None:
+        field, layer_depth = _remove_edge_nans(
+            field, "lat", layer_depth if "layer_depth" in locals() else None
+        )
+
+    # Regrid vertically
+    if depth is not None:
+        ds = xr.Dataset()
+        ds["s_rho"] = field["s_rho"]
+        vertical_regrid = VerticalRegridFromROMS(ds)
+        # Save attributes before vertical regridding
+        attrs = field.attrs
+        field = vertical_regrid.apply(field, layer_depth, np.array([depth])).squeeze()
+        # Reset attributes
+        field.attrs = attrs
+        title = title + f", depth = {depth}m"
+        depth_contours = False
+
+    # Plotting
+    kwargs = _set_plotting_kwargs(field, cmap_name)
+
+    if (eta is None and xi is None) and (lat is None and lon is None):
+        fig = plot_2d_horizontal_field(
+            field=field,
+            depth_contours=depth_contours,
+            with_dim_names=with_dim_names,
+            title=title,
+            kwargs=kwargs,
+            add_colorbar=add_colorbar,
+        )
+    else:
+        if len(field.dims) == 2:
+            if compute_interface_depth:
+                if max_nr_layer_contours:
+                    # restrict number of layer_contours to 10 for the sake of plot clearity
```

Comment:
This comment appears to be obsolete? The max layer contours parameter is variable, `interface_depth["s_w"]` is variable, but this states a fixed value of 10. The default is 10, but that isn't always the case.

Maybe this could state the intent (Avoid using contour counts that do not allow spacing) without a specific value, instead?